### PR TITLE
Move contact form PHP files

### PR DESCRIPTION
**Pull request by ice9js**
Part of #28405.

## Proposed changes:

This patch moves contact form PHP files from `plugins/jetpack/modules/contact-form` into `packages/forms/src/contact-form`.

I used this as an opportunity to organize the classes into separate files and move everything into object-oriented structures. There's still a lot of gradual improvements to be done but I think that's already a big step forward.  

**Note:** This PR will require #28417 to be merged, and the static file references to be updated before it can be fully tested.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pcsBup-N3-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

Use `composer phpunit` inside the package directory to run unit tests.

Enable the package on your jetpack install by using the `jetpack_contact_form_use_package` filter:
```
add_filter(
    'jetpack_contact_form_use_package',
    function () {
        return true;
    }
);
```

- Contact form module should load as expected.
- Verify that you can load the feedback area in the dashboard and the filters work as expected.
- Verify that you can export feedback as expected.
- Verify that the contact form gutenberg blocks load as expected in the editor.
- Verify that the contact form blocks render as expected on published pages or posts.

### CODE REVIEW

**Comment by ice9js** [comment]
Note to self: This currently includes all PHP changes from `projects/plugins/jetpack/modules/contact-form` up until `347af7955700f2df0c57c14c709c48b4cf6aaef5`. Any new commits will need to be merged in manually.

**Comment by ice9js** [comment]
I think I managed to move everything over and get it organized, as well as getting the old tests to pass. Unfortunately I had to disable four of them that rely on direct database calls or `get_posts` which we can't make work outside of WordPress'.

The code is still missing the changes from #27837 and #28640 but I think it's probably already more than big enough, so it'll be easier to move these over separately. Just keep in mind the new styling features might still be missing on this branch.

Other than that, please try to be thorough in your testing given this is a lot of code to move at once.

**Review by CGastrell — DISMISSED** [review]
Great job Kuba! Works as described, tested well on .com

File: projects/packages/forms/composer.json [hunk 1]
```diff
@@ -3,11 +3,10 @@
 	"description": "Jetpack Forms",
 	"type": "jetpack-library",
 	"license": "GPL-2.0-or-later",
-	"require": {},
 	"require-dev": {
 		"yoast/phpunit-polyfills": "1.0.4",
 		"automattic/jetpack-changelogger": "@dev",
-		"automattic/wordbless": "dev-master"
+		"automattic/wordbless": "^0.4.1"
```
Comment (jeherve) [inline projects/packages/forms/composer.json:10]: Should we use `@dev` here as well?

```suggestion
		"automattic/wordbless": "@dev"
```

File: projects/packages/forms/src/contact-form/class-admin.php [hunk 2]
```diff
@@ -0,0 +1,1481 @@
...
+				sanitize_file_name( $source )
+			);
+	}
+
+	/**
+	 * Build contact form button.
+	 *
+	 * @return void
+	 */
+	public function grunion_media_button() {
```
Comment (jeherve) [inline projects/packages/forms/src/contact-form/class-admin.php:412]: We'll probably need to add a check in a future PR:

Screenshot 2023-01-31 at 12 53 46

**Review by jeherve — APPROVED** [review]
Let's merge this for now! 🚢 I fixed some fatals and have a comment below, but nothing blocking imo.

**Pull request merged**